opensrp-reveal/src/test/java/org/smartregister/reveal/presenter/ListTaskPresenterTest.java

```diff
@@ -361,5 +405,326 @@ public void testSearchTasks() throws JSONException {
+        assertEquals(expectedCardDetails.getStatus(), actualCardDetails.getStatus());
+        assertEquals(expectedCardDetails.getDateCreated(), actualCardDetails.getDateCreated());
+        assertEquals(expectedCardDetails.getOwner(), actualCardDetails.getOwner());
+
+    }
+
+    @Test
+    public void testOnFamilyCardDetailsFetched() {
+
+        FamilyCardDetails expectedCardDetails = new FamilyCardDetails(COMPLETE, "1582279044", "nifi-user");
+
+        listTaskPresenter.onCardDetailsFetched(expectedCardDetails);
+
+        verify(listTaskView).openCardView(cardDetailsArgumentCaptor.capture());
+
+        FamilyCardDetails actualCardDetails = (FamilyCardDetails) cardDetailsArgumentCaptor.getValue();
+
+        assertEquals(COMPLETE, actualCardDetails.getStatus());
+        assertEquals("nifi-user", actualCardDetails.getOwner());
+        assertEquals("19 Jan 1970", actualCardDetails.getDateCreated());
+
+
+    }
+
+    @Test
+    public void testOnSprayCardDetailsFetched() {
+
+        SprayCardDetails expectedCardDetails = new SprayCardDetails(NOT_SPRAYED, "Residential", "2014-07-04T12:08:56.235-0700", "gideon", "Mark", "Available");
+
+        listTaskPresenter.onCardDetailsFetched(expectedCardDetails);
+
+        verify(listTaskView).openCardView(cardDetailsArgumentCaptor.capture());
+
+        SprayCardDetails actualCardDetails = (SprayCardDetails) cardDetailsArgumentCaptor.getValue();
+
+        assertEquals(NOT_SPRAYED, actualCardDetails.getStatus());
+        assertEquals("Residential", actualCardDetails.getPropertyType());
+        assertEquals("04 Jul 2014", actualCardDetails.getSprayDate());
+        assertEquals("gideon", actualCardDetails.getSprayOperator());
+        assertEquals("Mark", actualCardDetails.getFamilyHead());
+        assertEquals("Available", actualCardDetails.getReason());
+
+    }
+
+    @Test
+    public void testOnMosquitoHarvestCardDetailsFetched() {
+
+        IRSVerificationCardDetails expectedCardDetails = new IRSVerificationCardDetails(NOT_VISITED,
+                "yes", "no", "sprayed", "No chalk",
+                "No sticker", "No card");
+
+
+        listTaskPresenter.onCardDetailsFetched(expectedCardDetails);
+
+        verify(listTaskView).openCardView(cardDetailsArgumentCaptor.capture());
+
+        IRSVerificationCardDetails actualCardDetails = (IRSVerificationCardDetails) cardDetailsArgumentCaptor.getValue();
+
+        assertEquals(NOT_VISITED, actualCardDetails.getStatus());
+        assertEquals("yes", actualCardDetails.getTrueStructure());
+        assertEquals("no", actualCardDetails.getEligStruc());
+        assertEquals("sprayed", actualCardDetails.getReportedSprayStatus());
+        assertEquals("No chalk", actualCardDetails.getChalkSprayStatus());
+        assertEquals("No sticker", actualCardDetails.getStickerSprayStatus());
+        assertEquals("No card", actualCardDetails.getCardSprayStatus());
+    }
+
+    @Test
+    public void testOnIRSVerificationCardDetailsFetched() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/OpenSRP/opensrp-client-reveal/pullRequest?prid=4997357)